TileDB.Cloud/Udf.cs

```diff
@@ -0,0 +1,479 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Polly;
+using TileDB.Cloud.Rest.Api;
+using TileDB.Cloud.Rest.Model;
+
+namespace TileDB.Cloud
+{
+    public partial class RestUtil
+    {
+        public static class Udf
+        {
+            public static readonly UdfApi UdfApi;
+
+            static Udf()
+            {
+                UdfApi = Client.GetInstance().GetUdfApi();
+            }
+
+            #region Execute
+
+            /// <summary>
+            /// Execute a generic UDF
+            ///
+            /// If no `chargedOrg` is provided we will charge the current user's default charged namespace.
+            /// This defaults to the current username, but this can be reconfigured on TileDB Cloud Console
+            /// </summary>
+            /// <param name="udfUri">TileDB uri for UDF to execute</param>
+            /// <param name="args">JSON formatted arguments to pass to UDF</param>
+            /// <param name="chargedOrg">Organization to charge for UDF execution</param>
+            /// <returns>Stream containing UDF result</returns>
+            public static Stream ExecGeneric(
+                string udfUri, string? args = null, string? chargedOrg = null)
+            {
+                var genericUdf = new GenericUDF(udfInfoName: udfUri.Replace("tiledb://", ""), argument: args);
+                var policyResult = Client.GetInstance().GetRetryPolicyWrap().ExecuteAndCapture(
+                    () => UdfApi.SubmitGenericUDF(chargedOrg ?? GetDefaultChargedNamespace(), genericUdf)
+                );
+                CheckPolicyResult(policyResult);
```

Comment:
What does `CheckPolicyResult` do, and what happens if it fails? No need to change now, but it feels slightly clunky/repetitive so maybe we can encapsulate the error handling and unwrapping here in the future.

Reply:
I originally needed this function to use as a callback for checking the final result of asynchronous requests. It logs the error message to the console on failure and does nothing on success. 

```C#
private static void CheckPolicyResult(PolicyResult policyResult)
{
    if (policyResult.FinalException != null)
    {
        Console.WriteLine("Caught final exception! {0}", policyResult.FinalException.Message);
    }
}
```
 
Here's an example of using this as a continuation for async tasks. Without the continuation, a final error message isn't printed to the console.
 
 ```C#
 public static Task<PolicyResult<Stream>> ApplyAsync(...) {
     // ...
     var policyResult = Client.GetInstance().GetRetryAsyncPolicyWrap().ExecuteAndCaptureAsync(
        () => Udf.UdfApi.SubmitUDFAsync(arrayNamespace, arrayName, udf,
            chargedOrg ?? GetDefaultChargedNamespace()));
    policyResult.ContinueWith(t => CheckPolicyResult(t.Result));
 }
 ```

I thought about making this throw, but if I make the UDF API throw on failure I would also probably update existing functions to throw so they are all consistent. This PR was getting kinda large, so I thought I should focus on getting UDFs in before refactoring exception handling.

In new functions I wrote for this PR I also used it as a replacement for the same pattern that was in `RestUtil.cs` for synchronous requests like [CreateFile](https://github.com/TileDB-Inc/TileDB-Cloud-CSharp/blob/smr/sc-19991/example-udf/TileDB.Cloud/RestUtil.cs#L24).

Reply:
Sounds good, thanks. We want users to be able to write applications against this API as if they were calling a local function, so I think it makes sense to throw (either immediately or when the future is retrieved), but it's worth discussing further if anyone thinks that would not be idiomatic for C# (I don't know). In the Cloud-Py API we re-throw the exception that comes back from the cloud side to provide full context for understanding failures.